docs/inputgroup/datepicker-range.yml

```diff
@@ -0,0 +1,60 @@
+name: Date Picker - Range
+description: A date picker uses the input group component to display a field with a button next to it
+markup: |
+  <div aria-invalid="false" class="spectrum-InputGroup spectrum-Datepicker spectrum-Datepicker--range" role="combobox" aria-expanded="false" aria-haspopup="dialog">
+    <input type="text" class="spectrum-Textfield spectrum-InputGroup-field" placeholder="mm/dd/yyyy" name="start" value="">
+    <div class="spectrum-Datepicker--rangeDash"></div>
+    <input type="text" class="spectrum-Textfield spectrum-InputGroup-field" placeholder="mm/dd/yyyy" name="end" value="">
+    <button type="button" class="spectrum-FieldButton" tabindex="-1" aria-expanded="false" aria-haspopup="dialog" aria-label="Calendar">
+      <svg class="spectrum-Icon spectrum-UIIcon-ChevronDownMedium spectrum-Dropdown-icon" focusable="false" aria-hidden="true">
+        <use xlink:href="#spectrum-css-icon-ChevronDownMedium" />
+      </svg>
+    </button>
+    <!-- To render focus ring around entire input group when one of the inputs has keyboard focus, we need this: -->
+    <div class="spectrum-Datepicker-focusRing" role="presentation"></div>
+  </div>
+
+  <p/>
+
+  <div aria-invalid="true" class="spectrum-InputGroup spectrum-Datepicker spectrum-Datepicker--range is-invalid" role="combobox" aria-expanded="false" aria-haspopup="dialog">
+    <input type="text" class="spectrum-Textfield spectrum-InputGroup-field" aria-invalid="true" placeholder="mm/dd/yyyy" name="start" value="">
+    <div class="spectrum-Datepicker--rangeDash"></div>
+    <input type="text" class="spectrum-Textfield spectrum-InputGroup-field is-invalid" aria-invalid="true" placeholder="mm/dd/yyyy" name="end" value="">
+    <button type="button" class="spectrum-FieldButton is-invalid" tabindex="-1" aria-expanded="false" aria-haspopup="dialog" aria-label="Calendar">
+      <svg class="spectrum-Icon spectrum-UIIcon-ChevronDownMedium spectrum-Dropdown-icon" focusable="false" aria-hidden="true">
+        <use xlink:href="#spectrum-css-icon-ChevronDownMedium" />
+      </svg>
+    </button>
+    <!-- To render focus ring around entire input group when one of the inputs has keyboard focus, we need this: -->
+    <div class="spectrum-Datepicker-focusRing" role="presentation"></div>
```

Comment:
Mmmm, this can't be done with `:before`/`:after`? I don't see anything using those pseudo-elements right now.

Reply:
@majornista oh, I see -- it's being selected with the sibling selector, so we can't use pseudo-elements. Perhaps, instead, we should be programmatically applying `.is-focused` to the outer `.spectrum-Datepicker` and using `:focus-within` (for futureproofing)

Reply:
Only with additional javascript to check for `.focus-ring` className on a descendant element. And, we'd probably need an extra `.focus-ring` state, to distinguish between mouse a keyboard focus. With the added element, the style can be applied with pure CSS.

Reply:
Got it, so we would need to know "is a child element keyboard focused?" in JavaScript, and a separate class for keyboard vs mouse focus. @GarthDB thoughts on this? This is the first component that has anything like this.

Reply:
I prefer pure CSS for accessibility things like keyboard focus. It would be preferable to implement it with the same strategy as all the others, but I'm not seeing an obvious way to do that.
We could add something to the focus-ring polyfill, but that would kind of defeat the point of a standards-based polyfill.
The focus-withing strategy would be nice, it just means we're adding another polyfill requirement to focus. What would you prefer @lazd?

Reply:
Though I would prefer the polyfill, I don't want to hold up this PR. I agree with your sentiments and suppose I'm fine with the current approach.

Reply:
> Got it, so we would need to know "is a child element keyboard focused?" in JavaScript, and a separate class for keyboard vs mouse focus. @GarthDB thoughts on this? This is the first component that has anything like this.

The Rating component has a similar issue with how the focus outline is applied to the wrapper element. Devon was somewhat resistant to the approach of evaluating for descendant input  with `.focus-ring`. The difference for the Datepicker is that text inputs show blue border for mouse focus and border with box-shadow for keyboard focus.